manager/controllers/package_versions.go

```diff
@@ -0,0 +1,100 @@
+package controllers
+
+import (
+	"app/base/database"
+	"app/base/utils"
+	"app/manager/middlewares"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+	"net/http"
+)
+
+var PackageVersionFields = database.MustGetQueryAttrs(&PackageVersionItem{})
+var PackageVersionSelect = database.MustGetSelect(&PackageVersionItem{})
+var PackageVersionsOpts = ListOpts{
+	Fields: PackageVersionFields,
+	// By default, we show only fresh systems. If all systems are required, you must pass in:true,false filter into the api
+	DefaultFilters: map[string]FilterData{},
+	DefaultSort:    "evra",
+	SearchFields:   []string{"p.evra"},
+}
+
+type PackageVersionItem struct {
+	Evra string `json:"evra" csv:"evra" query:"evra" gorm:"column:evra"`
+}
+
+type PackageVersionsResponse struct {
+	Data  []PackageVersionItem `json:"data"`
+	Links Links                `json:"links"`
+	Meta  ListMeta             `json:"meta"`
+}
+
+func packagesNameID(pkgName string) *gorm.DB {
+	return database.Db.Table("package_name pn").
+		Where("pn.name = ?", pkgName)
+}
+
+func packageVersionsQuery(acc int, packageNameIDs []int) *gorm.DB {
+	query := database.SystemPackages(database.Db, acc).
+		Select(PackageVersionSelect).
+		Distinct("p.evra").
+		Where("sp.stale = false").
+		Where("spkg.name_id in (?)", packageNameIDs)
+	return query
+}
+
+//nolint: dupl
+// @Summary Show me all package versions installed on some system
+// @Description Show me all package versions installed on some system
+// @ID packageVersions
+// @Security RhIdentity
+// @Accept   json
+// @Produce  json
+// @Param    limit          query   int     false   "Limit for paging, set -1 to return all"
+// @Param    offset         query   int     false   "Offset for paging"
+// @Param    package_name    path    string    true  "Package name"
+// @Param    tags            query   []string  false "Tag filter"
+// @Param    filter[system_profile][sap_system]   query  string  false "Filter only SAP systems"
+// @Param    filter[system_profile][sap_sids][in] query []string  false "Filter systems by their SAP SIDs"
```

Comment:
Do we really need to have this filtering level here ?

Reply:
I think we need the same filtering level as used for package_systems to show relevant versions when filters are applied.

Reply:
Ok, removing filters and tags support here. I will that in a separate commit so we can easily add it back if we will need that.